services/src/datasets/postgres.rs

```diff
@@ -507,13 +508,21 @@ where
 
         let typed_meta_data = meta_data.to_typed_metadata();
 
-        let conn = self.conn_pool.get().await?;
+        let mut conn = self.conn_pool.get().await?;
 
-        // unique constraint on `id` checks if dataset with same id exists
+        let tx = conn.build_transaction().start().await?;
 
-        let stmt = conn
-            .prepare(
-                "
+        let existing_dataset = tx
+            .query_opt(
+                "SELECT TRUE FROM datasets WHERE name = $1::\"DatasetName\";",
+                &[&name],
+            )
+            .await?;
```

Comment:
Isn't there a return code when inserting that does the same thing? I doubt there is the need for a second query.

Reply:
you mean like here: https://github.com/geo-engine/geoengine/blob/780870522d26c485982545b796c9468d54f894e8/services/src/pro/users/postgres_userdb.rs#L52-L59 ?

would also be possible but if we ever have another Unique constraint on the dataset, we wouldn't be able to know which one was violated. So maybe it's not a good idea to use this technique after all?

Reply:
I guess we should also get that:

> For some types of errors, the server reports the name of a database object (a table, table column, data type, or constraint) associated with the error; for example, the name of the unique constraint that caused a unique_violation error. Such names are supplied in separate fields of the error report message so that applications need not try to extract them from the possibly-localized human-readable text of the message. As of PostgreSQL 9.3, complete coverage for this feature exists only for errors in SQLSTATE class 23 (integrity constraint violation), but this is likely to be expanded in future.

Reply:
d7e292016b37acad6e6578bbde878fb19c48c71d